src/adapter/src/catalog/transact.rs

```diff
@@ -684,13 +684,18 @@ impl Catalog {
                 )
                 .collect();
 
-                let (database_id, _) =
-                    tx.insert_user_database(&name, owner_id, database_privileges.clone())?;
+                let (database_id, _) = tx.insert_user_database(
+                    &name,
+                    owner_id,
+                    database_privileges.clone(),
+                    state.get_temporary_oids().collect(),
```

Comment:
Might be nice to take a `&BTreeSet<u32>` so we don't have to recreate the `BTreeSet` for each call. Also since we're never iterating over the set, a `HashSet` might be faster? Neither are a big deal though

Reply:
Done